examples/secrets-management-with-csi-secrets-driver-aws/main.tf

```diff
@@ -0,0 +1,268 @@
+  source         = "../../modules/kubernetes-addons"
+  eks_cluster_id = module.eks-blueprints.eks_cluster_id
+
+  # EKS Managed Add-ons
+  enable_amazon_eks_vpc_cni    = true
+  enable_amazon_eks_coredns    = true
+  enable_amazon_eks_kube_proxy = true
+
+  #K8s Add-ons
+  enable_secrets_store_csi_driver       = true
+  enable_csi_secrets_store_provider_aws = true
+
+  depends_on = [module.eks-blueprints.managed_node_groups]
+}
+
+
+data "aws_partition" "current" {}
+
+data "aws_caller_identity" "current" {}
+
+data "aws_eks_cluster" "eks_cluster" {
+  name = module.eks-blueprints.eks_cluster_id
+}
+
+#---------------------------------------------------------------
+# Parsing the config file to local and also extracting the ARNs of Secret Object
+#---------------------------------------------------------------
+locals {
+  secretconfig   = templatefile("${path.module}/secretconfig.yaml", {})
+  all_secret_arn = [for objects in yamldecode(local.secretconfig) : objects["objectName"]]
+}
+
+#---------------------------------------------------------------
+# Creating IAM Policy to be attached to the IRSA Role
+#---------------------------------------------------------------
+
+resource "aws_iam_policy" "this" {
+  description = "Sample application IAM Policy for IRSA"
+  name        = "${module.eks-blueprints.eks_cluster_id}-${var.application}-irsa"
+  policy      = data.aws_iam_policy_document.secrets_management_policy.json
+}
+
+#---------------------------------------------------------------
+# Creating IAM Role for Service Account
+#---------------------------------------------------------------
+
+module "iam_role_service_account" {
+  source = "../../modules/irsa"
+  addon_context = {
+    aws_caller_identity_account_id = data.aws_caller_identity.current.account_id
+    aws_caller_identity_arn        = data.aws_caller_identity.current.arn
+    aws_eks_cluster_endpoint       = data.aws_eks_cluster.eks_cluster.endpoint
+    aws_partition_id               = data.aws_partition.current.partition
+    aws_region_name                = data.aws_region.current.name
+    eks_cluster_id                 = module.eks-blueprints.eks_cluster_id
+    eks_oidc_issuer_url            = module.eks-blueprints.eks_oidc_issuer_url
+    eks_oidc_provider_arn          = "arn:${data.aws_partition.current.partition}:iam::${data.aws_caller_identity.current.account_id}:oidc-provider/${module.eks-blueprints.eks_oidc_issuer_url}"
+    tags                           = {}
+  }
+  kubernetes_namespace       = var.application
+  kubernetes_service_account = "${var.application}-sa"
+  irsa_iam_policies          = [aws_iam_policy.this.arn]
+
+  depends_on = [module.eks-blueprints]
+}
+
+#---------------------------------------------------------------
+# Kubernetes CRD to create the "SecretProviderClass" to represent the Secrets Manager Secrets
+#---------------------------------------------------------------
+
+resource "kubernetes_manifest" "csi_secrets_store_crd" {
```

Comment:
we should use [`kubectl_manifest`](https://registry.terraform.io/providers/gavinbunney/kubectl/latest/docs/resources/kubectl_manifest) in the examples to avoid hitting this issue when deploying cluster and addons https://github.com/hashicorp/terraform-provider-kubernetes/issues/1453